### PR TITLE
Add REST examples

### PR DESCRIPTION
**Pull request by tmccrmck**
## Checklist

I have:

- [x] read and followed the contributing guidelines
- [x] ran `python readme.py` after making changes (per contributing guidelines)
- [x] added required testing (per contributing guidelines)

## Changes
- Adds two REST scripts
- Mimics cli folder structure

### CODE REVIEW

File: cli/how-to-deploy-rest.sh [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+# define how to wait
+wait_for_completion () {
+    operationid=$(echo $1 | grep -Fi Azure-AsyncOperation | sed "s/azure-asyncoperation: //" | tr -d '\r')
+    # TODO error handling here
+    operation_status="unknown"
+
+    while [[ $operation_status != "Succeeded" && $operation_status != "Failed" ]]
+    do
+        $echo "Getting operation status from: $operationid"
+        operation_result=$(curl --location --request GET $operationid --header "Authorization: Bearer $TOKEN")
```
Comment (tmccrmck) [inline cli/how-to-deploy-rest.sh:46]: Unfortunately, MIR can take so long to compute that this token becomes invalid

File: cli/how-to-deploy-rest.sh [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+# delete endpoint
+curl --location --request DELETE "https://management.azure.com/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE_NAME/onlineEndpoints/my-endpoint?api-version=$API_VERSION" \
+--header "Content-Type: application/json" \
+--header "Authorization: Bearer $TOKEN"
+
+# TODO: we can get the default container from listing datastores
+# TODO using the latter two as env vars shouldn't be necessary
+az storage blob upload-batch -d $AZUREML_DEFAULT_CONTAINER/score \
+ -s endpoints/online/model-1/onlinescoring --account-name $AZURE_STORAGE_ACCOUNT --account-key $AZURE_STORAGE_KEY
```
Comment (tmccrmck) [inline cli/how-to-deploy-rest.sh:62]: According to the docs, `$AZURE_STORAGE_ACCOUNT` and `$AZURE_STORAGE_KEY` shouldn't be necessary if they are environment variables but I've never been able to get this to work on Linux

File: cli/how-to-deploy-rest.sh [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+# <create environment variables>
+SUBSCRIPTION_ID="7ab7d5bc-5d9e-47ef-80e6-2dffa8ca83a1"
+RESOURCE_GROUP="trmccorm-centraluseuap"
+WORKSPACE="trmccorm-centraluseuap"
+API_VERSION="2021-03-01-preview"
+
+TOKEN=$(az account get-access-token | jq -r ".accessToken")
+
+AZURE_STORAGE_ACCOUNT="trmccormcentra1277620275"
```
Comment (rsethur) [inline cli/how-to-deploy-rest.sh:18]: Is there a way to retrieve these 3 variables (associated with workspace) instead of having user specify them?

Comment (tmccrmck) [reply cli/how-to-deploy-rest.sh:18]: Not that I know of

**Pull request merged**